### PR TITLE
[FLINK-20022][docs] Move statebackend tradeoffs out of checklist

### PR DESCRIPTION
**Pull request by zentol**
The production readiness checklist contains more information for which statebackend one should use than the actual statebackend page, amusingly linking to the statebackend docs for pros/cons despite being more explicit about them.
Move this information to the statebackend page instead.

### CODE REVIEW

**Comment by StephanEwen** [comment]
I agree the section belongs to the state backend docs.

It would be good, though, to have a pointer in the Production-ready Checklist, like "Choose the right State Backend for your use case" with a link to the full text in the state backend docs. It is a critical config choice after all.

File: docs/ops/production_ready.md [hunk 1]
```diff
@@ -52,18 +52,6 @@ By default, operator uids are generated by traversing the JobGraph and hashing s
 While this is comfortable from a user perspective, it is also very fragile, as changes to the JobGraph (e.g., exchanging an operator) results in new UUIDs.
 To establish a stable mapping, we need stable operator uids provided by the user through `setUid(String uid)`.
 
-### Choose The Right State Backend
-
-Currently, Flink's savepoint binary format is state backend specific.
-A savepoint taken with one state backend cannot be restored using another, and you should carefully consider which backend you use before going to production.
```
Comment (tillrohrmann) [inline docs/ops/production_ready.md:58]: I know this is not related to this PR but is this actually still true? @carp84? If not, then we should probably update it.

File: docs/ops/production_ready.md [hunk 2]
```diff
@@ -52,18 +52,6 @@ By default, operator uids are generated by traversing the JobGraph and hashing s
 While this is comfortable from a user perspective, it is also very fragile, as changes to the JobGraph (e.g., exchanging an operator) results in new UUIDs.
 To establish a stable mapping, we need stable operator uids provided by the user through `setUid(String uid)`.
 
-### Choose The Right State Backend
```
Comment (tillrohrmann) [inline docs/ops/production_ready.md:55]: I agree with Stephan that we should keep a link to the state backend page here.

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @zentol. The change looks good to me. I also agree with Stephan to keep a link to the state backend page from the production readiness page. I will add it before merging this PR.

File: docs/ops/production_ready.md (hunk 1 above)
Comment (carp84) [reply docs/ops/production_ready.md:58]: Sorry for the late reply. And yes, this is still true. The efforts of unifying savepoint binary format and supporting state backend switch is tracked by [FLIP-41](https://cwiki.apache.org/confluence/display/FLINK/FLIP-41%3A+Unify+Binary+format+for+Keyed+State) but not done yet.

Comment (tillrohrmann) [reply docs/ops/production_ready.md:58]: Ok, thanks for the confirmation @carp84.

**Pull request closed without merging**